src/irr/asset/CImageLoaderOpenEXR.cpp

```diff
@@ -29,12 +29,65 @@ namespace irr
 			if (!_file)
 				return {};
 
+			SContext ctx;
+			const auto& fileName = _file->getFileName().c_str();
+
+			if (!readMagicNumber())
+				return {};
+
+			readVersionField();
+			readHeader(fileName, ctx);
+
+			Array2D<Rgba> pixels;
+			int width;
+			int height;
+
+			readRgba(fileName, pixels, width, height);
+
+			// TODO put data into buffer, consider OpenEXR 2.0 supported saving layouts 
```

Comment:
what are `layouts` ?

Reply:
I meant I need to support tiled files, deep imagesm multilayered images and all stuff I guess you would like it to support:)

Reply:
it only needs to load regular OpenEXR images produced by mitsuba, and GIMP, you can error out on the more exotic stuff.